internal/cmd/controller/gitops/reconciler/status_controller.go

```diff
@@ -165,23 +190,80 @@ func setStatus(list *v1alpha1.BundleDeploymentList, gitrepo *v1alpha1.GitRepo) e
 	return nil
 }
 
+// setReadyStatusFromBundle fetches all bundles from a given gitrepo, checks the ready status conditions
+// from the bundles and applies one on the gitrepo if it isn't ready. The purpose is to make
+// rendering issues visible in the gitrepo status. Those issues need to be made explicitly visible
+// since the other statuses are calculated from bundle deployments, which do not exist when
+// rendering manifests fail. Should an issue be on the bundle, it will be copied to the gitrepo.
+func (r StatusReconciler) setReadyStatusFromBundle(ctx context.Context, gitrepo *fleet.GitRepo) error {
+	bList := &fleet.BundleList{}
+	err := r.List(ctx, bList, client.MatchingLabels{
+		fleet.RepoLabel: gitrepo.Name,
+	}, client.InNamespace(gitrepo.Namespace))
+	if err != nil {
+		return err
+	}
+
+	found := false
+	// Find a ready status condition in a bundle which is not ready.
+	var condition genericcondition.GenericCondition
+bundles:
+	for _, bundle := range bList.Items {
+		if bundle.Status.Conditions == nil {
+			continue
+		}
+
+		for _, c := range bundle.Status.Conditions {
+			if c.Type == string(fleet.Ready) && c.Status == v1.ConditionFalse {
+				condition = c
+				found = true
+				break bundles
+			}
+		}
+	}
+
+	// No ready condition found in any bundle, nothing to do here.
+	if !found {
+		return nil
+	}
+
+	found = false
+	newConditions := make([]genericcondition.GenericCondition, 0, len(gitrepo.Status.Conditions))
+	for _, c := range gitrepo.Status.Conditions {
+		if c.Type == string(fleet.Ready) {
+			// Replace the ready condition with the one from the bundle
+			newConditions = append(newConditions, condition)
+			found = true
+			continue
+		}
+		newConditions = append(newConditions, c)
+	}
+	if !found {
+		// Add the ready condition from the bundle to the gitrepo.
+		newConditions = append(newConditions, condition)
+	}
```

Comment:
Just curious: would readiness of gitrepos be previously computed from bundle deployments only, then?

Reply:
Yes. This is how it was:
```mermaid
flowchart LR
  GR[GitRepo]
  BD[Bundle Deployment]
  B[Bundle]

  GR -->|"`fetches the status of and
  aggregates the status of`"| BD
  GR -->|creates| B
  B -->|creates one or more and
  fetches the status of| BD
```
With this implementation it becomes:
```mermaid
flowchart LR
  GR[GitRepo]
  BD[Bundle Deployment]
  B[Bundle]

  GR -->|"`fetches the status of and
  aggregates the status of`"| BD
  GR -->|creates| B
  B -->|creates one or more and
  fetches the status of| BD

  GR ==>|needs to fetch
  the status of| B
```
And how I thought it perhaps could be, but I did not implement it:
```mermaid
flowchart LR
  GR[GitRepo]
  BD[Bundle Deployment]
  B[Bundle]

  GR -->|"`creates and fetches the
  (aggregated) status of`"| B
  B -->|creates one or more and
  fetches the status of| BD
```

Reply:
Nice, thanks for clarifying, cool diagrams too!